### PR TITLE
Updates to dependencies and test workflows

### PR DESCRIPTION
**Pull request by gaby**
- Remove support for `go 1.16` from all storage drivers
- Add `go 1.20` to test matrix of each storage driver
- Add `macos-latest` to test matrix of each storage driver
- Run `go mod tidy` on each driver
- Added matrix test to vulnerability and security workflows
- Fixed the platform field for each workflow (was hardcoded to ubuntu-latest)

### CODE REVIEW

File: .github/workflows/vulncheck.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
     - name: Install Go
       uses: actions/setup-go@v4
       with:
-        go-version: 1.19.x
+        go-version: 1.18.x
```
[the remark this answers is not in the document]
Comment (gaby) [reply .github/workflows/vulncheck.yml:10]: True! Adding it in a second

**Comment by sixcolors** [comment]
@gaby looks good to me so far. But, I’m getting blurry eyed. I’ll run tests locally tomorrow just to see if anything odd comes up on actual macOS vs macOS-latest in the ci.

Thanks for the PR!

**Comment by gaby** [comment]
I just realized... These test are not actually running on each OS. They all have the same "runs-on" hardcoded to ubuntu-latest

**Comment by gaby** [comment]
`gosec` requires go 1.19 to run

**Comment by gaby** [comment]
> > I just realized... These test are not actually running on each OS. They all have the same "runs-on" hardcoded to ubuntu-latest
> 
> told you I’m getting blurry eyed. 🥱🛌
> 
> runs-on: ${{ matrix.platform }}

We are running Docker containers for several test, and these are only support on Linux

**Comment by gaby** [comment]
Don't merge yet, I forgot to remove the platforms from each test. Most of the tests can only run in "ubuntu-latest".

This will reduce the number of jobs created by the CI

**Comment by sixcolors** [comment]
> Awesome, I will add those then. Removing the platforms reduce the number of tests down to less than 40

The same guy did a bunch, I guess he needed them for the same reason:

https://github.com/ankane/setup-mysql#related-actions

Anyway, I'm happy with the PR. Did you want me to approve or wait for more commits?

**Review by sixcolors — APPROVED** [review]
LGTM

**Comment by gaby** [comment]
@sixcolors Discussing on discord what to do about the missing platforms.

**Pull request merged**